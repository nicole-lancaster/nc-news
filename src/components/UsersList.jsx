import SingleUser from "./SingleUser.jsx";
import { useEffect, useState } from "react";
import { fetchUsers } from "../api.js";
import { NewspaperIcon } from "@heroicons/react/24/outline";

const UsersList = ({ users, setUsers, currentUser, setCurrentUser }) => {
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    setIsLoading(true);
    fetchUsers(setUsers).then((usersList) => {
      setUsers(usersList);
      setIsLoading(false);
    });
  }, [setUsers]);

  if (isLoading)
    return (
      <div className="flex flex-col justify-center items-center animate-pulse">
        <NewspaperIcon className="h-8 w-auto mr-2 animate-pulse" />
        <p className="font-mono">Loading users...</p>
      </div>
    );

  return (
    <section className="flex flex-col">
      <div className="flex-grow">
        <ul className="flex flex-row flex-wrap justify-center items-center">
          {users.map((user) => {
            return (
              <SingleUser
                key={user.username}
                user={user}
                currentUser={currentUser}
                setCurrentUser={setCurrentUser}
              />
            );
          })}
        </ul>
      </div>
    </section>
  );
};

export default UsersList;
