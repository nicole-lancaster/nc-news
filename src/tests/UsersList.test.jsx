import { render, screen } from "@testing-library/react";
import UsersList from "../components/UsersList";
import { UserContext } from "../contexts/User";

jest.mock("../api.js", () => ({
  fetchUsers: () =>
    Promise.resolve([
      { name: "Rob", username: "robR" },
      { name: "Nicole", username: "nicoleL" },
    ]),
}));

describe("Users List component", () => {
  test("should match rendered snapshot", () => {
    const { asFragment } = render(
      <UsersList />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test("should display users", async () => {
    render(
      <UserContext.Provider
        value={{ currentUser: undefined, setCurrentUser: () => {} }}
      >
        <UsersList />
      </UserContext.Provider>
    );

    await screen.findByText("Rob as robR");
    await screen.findByText("Nicole as nicoleL");
  });
});
