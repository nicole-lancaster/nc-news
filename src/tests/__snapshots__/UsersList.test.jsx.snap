// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users List component should match rendered snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center items-center animate-pulse"
  >
    <svg
      aria-hidden="true"
      class="h-8 w-auto mr-2 animate-pulse"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 7.5h1.5m-1.5 3h1.5m-7.5 3h7.5m-7.5 3h7.5m3-9h3.375c.621 0 1.125.504 1.125 1.125V18a2.25 2.25 0 01-2.25 2.25M16.5 7.5V18a2.25 2.25 0 002.25 2.25M16.5 7.5V4.875c0-.621-.504-1.125-1.125-1.125H4.125C3.504 3.75 3 4.254 3 4.875V18a2.25 2.25 0 002.25 2.25h13.5M6 7.5h3v3H6v-3z"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
    <p
      class="font-mono"
    >
      Loading users...
    </p>
  </div>
</DocumentFragment>
`;
